File: Week1_PSETS/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_full_pyramid(int height);
void print_pyramid_spaces(int spaces);
void print_pyramid_hash(int length);

int main(void)
{
    int height;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 23);

    print_full_pyramid(height);
}

void print_full_pyramid(int height)
{
    int spaces_number = height;

    for (int i = 0; i < height; i++)
    {
        print_pyramid_spaces(spaces_number - 1);
        print_pyramid_hash(i + 1);
        printf("  ");
        print_pyramid_hash(i + 1);
        printf("\n");

        spaces_number--;
    }
}

void print_pyramid_spaces(int spaces)
{
    for (int i = spaces; i > 0; i--)
    {
        printf(" ");
    }
}

void print_pyramid_hash(int length)
{

    for (int i = 0; i < length; i++)
    {
        printf("#");
    }
}

File: Week6_PSETS/sentimental-mario-more/mario.py
from cs50 import get_int


def main():

    # Get input only between 1 - 8
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break

    recursiveHash(height, 0)


def recursiveHash(height, spaces):

    # Base case
    if height <= 0:
        return

    # Print pyramid of height - 1
    recursiveHash(height - 1, spaces + 1)

    # Print spaces
    for i in range(spaces):
        print(" ", end="")

    # Print first hashes
    for i in range(height):
        print("#", end="")

    # Print middle spaces
    print("  ", end="")

    # Print second hashes
    for i in range(height):
        print("#", end="")

    # Move to new line
    print("")


main()
